mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java

```diff
@@ -0,0 +1,209 @@
+package org.mlflow.tracking;
+
+import org.apache.http.client.utils.URIBuilder;
+
+import org.mlflow.api.proto.Service.*;
+import org.mlflow.tracking.creds.BasicMlflowHostCreds;
+import org.mlflow.tracking.creds.MlflowHostCredsProvider;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Client to an MLflow Tracking Sever.
+ */
+public class MlflowClient {
+  private static long DEFAULT_EXPERIMENT_ID = 0;
+
+  private final MlflowProtobufMapper mapper = new MlflowProtobufMapper();
+  private final MlflowHttpCaller httpCaller;
+
+  /** Returns a default client based on the MLFLOW_TRACKING_URI environment variable. */
+  public MlflowClient() {
+    this(getDefaultTrackingUri());
+  }
+
+  /** Instantiates a new client using the provided tracking uri. */
+  public MlflowClient(String trackingUri) {
+    this(getHostCredsProviderFromTrackingUri(trackingUri));
+  }
+
+  /**
+   * Creates a new MlflowClient. Users should prefer constructing ApiClients via
+   * {@link #MlflowClient()} ()} or {@link #MlflowClient(String)} if possible.
+   */
+  public MlflowClient(MlflowHostCredsProvider hostCredsProvider) {
+    httpCaller = new MlflowHttpCaller(hostCredsProvider);
+  }
+
+  /** Returns the run associated with the id. */
+  public Run getRun(String runUuid) {
+    URIBuilder builder = newURIBuilder("runs/get").setParameter("run_uuid", runUuid);
+    return mapper.toGetRunResponse(httpCaller.get(builder.toString())).getRun();
+  }
+
+  /** Creates a new run under the default experiment with no application name. */
+  public RunInfo createRun() {
+    return createRun(DEFAULT_EXPERIMENT_ID);
+  }
+
+  /** Creates a new run under the given experiment with no application name. */
+  public RunInfo createRun(long experimentId) {
+    return createRun(experimentId, "");
```

Comment:
Should we pass `""` or null? If the backend supports null, I think that's better. Otherwise I'd actually pass "Java Application" here until we can find a better default.

Reply:
Backend currently converts null to empty string, so will change to Java Application.